### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by SetMyDream**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/GenericDao.java [hunk 1]
```diff
@@ -3,11 +3,15 @@
 import com.dev.cinema.exceptions.DataProcessingException;
 import com.dev.cinema.util.HibernateUtil;
 import java.util.List;
+import org.apache.log4j.Logger;
 import org.hibernate.Session;
 import org.hibernate.Transaction;
 import org.hibernate.query.Query;
 
 public interface GenericDao<T> {
+
```
Comment (Vosker) [inline src/main/java/com/dev/cinema/dao/GenericDao.java]: empty space, its a bit awkward)

**Review by Vosker — COMMENTED** [review]
Ок. Well done. Мб добавь еще propetries логгера, чтобы менторы подсказали в случае чего.

File: src/main/java/com/dev/cinema/Main.java [hunk 2]
```diff
@@ -16,11 +16,15 @@
 import com.dev.cinema.service.UserService;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
+import org.apache.log4j.Logger;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java:19]: create `logs` folder with `.log` file in it, and change the path to the logs file in your `.properties` 
don't forget to add the folder to gitignore

File: src/main/java/com/dev/cinema/Main.java [hunk 3]
```diff
@@ -16,11 +16,15 @@
...
 
 public class Main {
     private static Injector injector = Injector.getInstance("com.dev.cinema");
 
+    private static final Logger log = Logger.getLogger(Main.class);
+
     public static void main(String[] args) throws AuthenticationException {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: don't separate fields with empty lines

also, empty lines after method signature are redundant

File: src/main/java/com/dev/cinema/Main.java [hunk 4]
```diff
@@ -66,32 +70,32 @@ public static void main(String[] args) throws AuthenticationException {
         authenticationService.register(johny.getEmail(), johny.getPassword());
         johny = authenticationService.login(johny.getEmail(), johny.getPassword());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: let's wrap call to `login()` method in try-catch and add log.warn with the informative message in `catch` block

File: src/main/java/com/dev/cinema/Main.java [hunk 5]
```diff
@@ -66,32 +70,32 @@ public static void main(String[] args) throws AuthenticationException {
...
         UserService userService = (UserService) injector.getInstance(UserService.class);
-        System.out.println(userService.findByEmail(johny.getEmail()).get());
+        log.info(userService.findByEmail(johny.getEmail()).get());
 
         User den = new User();
         den.setEmail("den@somemail.com");
         den.setPassword("dent");
         authenticationService.register(den.getEmail(), den.getPassword());
         den = authenticationService.login(den.getEmail(), den.getPassword());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/Main.java]: same here

File: src/main/java/com/dev/cinema/dao/GenericDao.java [hunk 6]
```diff
@@ -3,11 +3,15 @@
...
+    Logger logger = Logger.getLogger(GenericDao.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/GenericDao.java:12]: access modifiers

File: src/main/java/com/dev/cinema/exceptions/AuthenticationException.java [hunk 7]
```diff
@@ -1,11 +1,18 @@
...
 
+import org.apache.log4j.Logger;
+
 public class AuthenticationException extends Exception {
+
+    private static final Logger logger = Logger.getLogger(RuntimeException.class);
+
     public AuthenticationException(String message, Throwable cause) {
         super(message, cause);
+        logger.error(message,cause);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/exceptions/AuthenticationException.java]: this is redundant, let's remove logging from this class
exceptions are logged automatically

File: src/main/java/com/dev/cinema/exceptions/DataProcessingException.java [hunk 8]
```diff
@@ -1,7 +1,14 @@
...
+import org.apache.log4j.Logger;
+
 public class DataProcessingException extends RuntimeException {
+
+    private static final Logger logger = Logger.getLogger(RuntimeException.class);
+
     public DataProcessingException(String message, Throwable cause) {
         super(message, cause);
+
+        logger.error(message,cause);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/exceptions/DataProcessingException.java]: same

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
create logs folder with .log file in it, and change the path to the logs file in your .properties
don't forget to add the folder to gitignore

File: src/main/java/com/dev/cinema/dao/GenericDao.java (hunk 6 above)
Comment (SetMyDream) [reply src/main/java/com/dev/cinema/dao/GenericDao.java:12]: it`s interface, so it  not supposed to, isn`t it?

Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/dao/GenericDao.java:12]: you are right!

File: .gitignore [hunk 9]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
 target/*
+.logs/*
```
Comment (KseniiaMakarova) [inline .gitignore]:
```suggestion
logs/*
```

File: src/main/resources/log4j.properties [hunk 10]
```diff
@@ -9,7 +9,7 @@ log4j.appender.stdout.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1
 
 # Redirect log messages to a log file, support file rolling.
 log4j.appender.file=org.apache.log4j.RollingFileAppender
-log4j.appender.file.File=C:/Users/Artem/Documents/com.dev.cinema.logs/log4j-application.log
+log4j.appender.file.File=C:/Users/Artem/IdeaProjects/hibernate-cinema/logslog4j-application.log
```
Comment (KseniiaMakarova) [inline src/main/resources/log4j.properties]:
```suggestion
log4j.appender.file.File=C:/Users/Artem/IdeaProjects/hibernate-cinema/logs/log4j-application.log
```

File: src/main/java/com/dev/cinema/dao/GenericDao.java [hunk 11]
```diff
@@ -9,9 +9,7 @@
 import org.hibernate.query.Query;
 
 public interface GenericDao<T> {
-
-    Logger logger = Logger.getLogger(GenericDao.class);
-
+    public static final Logger logger = Logger.getLogger(GenericDao.class);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/GenericDao.java]: add an empty line between the field and the method

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**